### PR TITLE
Add `SINTER` command.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
https://redis.io/commands/sinter/

### CODE REVIEW

File: java/client/src/main/java/glide/api/BaseClient.java [hunk 1]
```diff
@@ -460,6 +461,11 @@ public CompletableFuture<Long> scard(@NonNull String key) {
         return commandManager.submitNewCommand(SCard, new String[] {key}, this::handleLongResponse);
     }
 
+    @Override
+    public CompletableFuture<Set<String>> sinter(@NonNull String[] keys) {
+        return commandManager.submitNewCommand(SInter, keys, this::handleSetResponse);
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/BaseClient.java:466]: Note: Let's ask about the return value, as redis documentation says that it returns an Array 🤨

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 2]
```diff
@@ -75,4 +75,19 @@ public interface SetBaseCommands {
      * }</pre>
      */
     CompletableFuture<Long> scard(String key);
+
+    /**
+     * Gets the intersection of all the given sets.<br>
+     * Missing set is interpreted as empty and causes an empty response.
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:81]: I think you could remove this line and add it  to the return.
     * If any set is missing it is interpreted as empty and causes an empty response.

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 3]
```diff
@@ -75,4 +75,19 @@ public interface SetBaseCommands {
...
+     *
+     * @see <a href="https://redis.io/commands/sinter/">redis.io</a> for details.
+     * @param keys The keys of the list.
+     * @return A <code>Set</code> of members which are present in all given sets.
+     * @example
+     *     <pre>{@code
+     * String[] values = client.sinter(new String[] {"list1", "list2"}).get();
+     * assert values.length > 1;
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:89]: Something more explanatory maybe or maybe add a mini comment next to it?

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 4]
```diff
@@ -1425,6 +1426,20 @@ public T type(@NonNull String key) {
         return getThis();
     }
 
+    /**
+     * Gets the intersection of all the given sets.<br>
+     * Missing set is interpreted as empty and causes an empty response.
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:1431]: Same comment as above.

**Review by SanHalacogluImproving — APPROVED** [review]

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java (hunk 2 above)
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/SetBaseCommands.java]: Fixed thanks

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java (hunk 3 above)
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/SetBaseCommands.java]: Added

File: java/client/src/main/java/glide/api/BaseClient.java (hunk 1 above)
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/BaseClient.java:466]: Tested with redis-cli - the doc is outdated.
RESP2 returns array, RESP3 returns a set
I added data convertion to glide-core which used with RESP2
```
$ redis-cli -3
127.0.0.1:6379> sinter s1
1~ "a"
127.0.0.1:6379> sinter s1
1~ "a"
127.0.0.1:6379> sinter s1 s2
(empty set)
127.0.0.1:6379> smembers s1
1~ "a"
127.0.0.1:6379>
```

**Review by jonathanl-bq — APPROVED** [review]

**Pull request merged**